naming/src/main/java/org/jboss/as/naming/InitialContext.java

```diff
@@ -112,8 +115,17 @@ protected Context getDefaultInitCtx() throws NamingException {
             } else {
                 final ClassLoader classLoader = WildFlySecurityManager.getCurrentContextClassLoaderPrivileged();
                 try {
-                    final Class<?> factoryClass = Class.forName(factoryClassName, true, classLoader);
-                    defaultInitCtx = ((javax.naming.spi.InitialContextFactory)factoryClass.newInstance()).getInitialContext(myProps);
+                    final Constructor<javax.naming.spi.InitialContextFactory> factoryClassConstructor;
+                    if (WildFlySecurityManager.isChecking()) {
+                        factoryClassConstructor = AccessController.doPrivileged((PrivilegedExceptionAction<Constructor<javax.naming.spi.InitialContextFactory>>) () -> {
+                            final Class<?> factoryClass = Class.forName(factoryClassName, false, classLoader);
```

Comment:
It is not enough to request changes on this PR but IMO I find it better to split all of the common code out, in this case it could just be a static method.

Where code like this is duplicated there is a risk in the future that the two get out of sync so bugs only occur in one case or the other.